gcloud/src/main/java/org/jimmutable/gcloud/search/IndexManager.java

```diff
@@ -0,0 +1,167 @@
+package org.jimmutable.gcloud.search;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.jimmutable.gcloud.logging.LogSupplier;
+
+import com.google.appengine.api.search.Document;
+import com.google.appengine.api.search.Field;
+import com.google.appengine.api.search.Index;
+import com.google.appengine.api.search.IndexSpec;
+import com.google.appengine.api.search.OperationResult;
+import com.google.appengine.api.search.PutException;
+import com.google.appengine.api.search.PutResponse;
+import com.google.appengine.api.search.Results;
+import com.google.appengine.api.search.ScoredDocument;
+import com.google.appengine.api.search.SearchException;
+import com.google.appengine.api.search.SearchServiceFactory;
+import com.google.appengine.api.search.StatusCode;
+
+/**
+ * Use this class to upsert, delete and get a document from a search Index
+ * 
+ * @author trevorbox
+ *
+ */
+public class IndexManager {
+
+	private static final Logger logger = Logger.getLogger(IndexManager.class.getName());
+
+	private static Index getIndex(IndexId indexId) {
+		IndexSpec indexSpec = IndexSpec.newBuilder().setName(indexId.getSimpleValue()).build();
+		return SearchServiceFactory.getSearchService().getIndex(indexSpec);
+	}
+
```

Comment:
Function name should be createIndex (since it creates and object, not a getter)

Also, you need validation on your params